src/components/EasyDropdown/EasyDropdown.tsx

```diff
@@ -1,30 +1,63 @@
-import React, { useCallback, useMemo, useState } from 'react';
-import PropTypes from 'prop-types';
+import React, { useCallback, useMemo, useState, ReactElement } from 'react';
 import classNames from 'classnames';
 
 import Button from '../Button';
 import Dropdown from '../Dropdown';
 
 import styles from './EasyDropdown.module.css';
+import { MenuProps } from '../Dropdown/Menu/Menu';
+import { DropdownProps } from '../Dropdown/Dropdown';
+import { ItemProps, ButtonElementProps } from '../Dropdown/Menu/Item/Item';
 
-const EasyDropdown = ({
-  children: initChildren,
-  className,
-  defaultIsOpen,
-  disabled,
-  isOpen: controlledIsOpen,
-  menuItems,
-  menuProps,
-  onToggle: parentOnToggle,
-  toggleProps,
+export type MenuItem = ItemProps & {
+  /** Children will be rendered instead of the label, if provided */
+  children?: React.ReactNode;
+  /** If you provide group IDs, the menu items will be grouped with dividers between them. */
+  group?: string;
+  /** This will be the array key and the fallback contents */
+  label: string;
+  onClick?: () => void;
+};
+
+interface EasyDropdownProps extends DropdownProps {
+  /** Pass a custom node if you want to control the toggle fully. */
+  children?: React.ReactNode;
+  /** Totally optional, for additional styling */
+  className?: string;
+  /** If defaultIsOpen is provided, the component will run in "uncontrolled" mode */
+  defaultIsOpen?: boolean;
+  disabled?: boolean;
+  /** Without `defaultIsOpen`, `isOpen` fully controls the dropdown state */
+  isOpen?: boolean;
+  /** An array of items that comprise the menu */
+  menuItems?: MenuItem[];
+  menuProps?: MenuProps;
+  /** Without `defaultIsOpen`, `onToggle` is the only way to set state. With it, it's a convenience callback. */
+  toggleProps?: React.ButtonHTMLAttributes<HTMLButtonElement>;
+}
+
+const EasyDropdown: React.FC<EasyDropdownProps> = ({
+  children: initChildren = undefined,
+  className = '',
+  defaultIsOpen = undefined,
+  disabled = false,
+  isOpen: controlledIsOpen = false,
+  menuItems = [],
+  menuProps = {
+    position: 'bottomRight',
+  },
+  onToggle: parentOnToggle = () => {},
+  toggleProps = {
+    className: '',
+  },
   ...passedProps
 }) => {
   const [uncontrolledIsOpen, setUncontrolledIsOpen] = useState(defaultIsOpen);
   const isControlled = defaultIsOpen === undefined;
   const isOpen = isControlled ? controlledIsOpen : uncontrolledIsOpen;
 
   // Group menu items while preserving their order.
-  const menuItemGroups = useMemo(
+  const menuItemGroups = useMemo<Record<string, MenuItem[]>>(
```

Comment:
Super minor: You _might_ not need to define the type here, since TS is sometimes pretty smart about inferring types.
I think it's worth trying, but not a big deal if TS demands you leave it in.

Reply:
Done.